File: Software/Surface/TableServer/TableServer/JsonObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Newtonsoft.Json;
namespace TableServer
{
    public class JsonObject
    {
        public enum Type { auth, data};
        
        public Type type {get;set;}
        public String data { get; set; }
        
        public JsonObject(Type type, String data)
        {
            this.data = data;
            this.type = type;
        }
        
        //Vor working convert in Newtonsoft.Json we have to use the c# construct of getter and setter
        /*public String getDataString() { return data; }
        public Type getType() { return type; }

        public void setType(Type type) { this.type = type;  }
        public void setData(String data) { this.data = data; }*/

    }
}

File: Software/Tablet/LeviTabLib/src/com/JsonObject.java
package com;

import org.json.JSONException;
import org.json.JSONObject;

public class JsonObject {
	public enum Type {
		auth, data
	};

	private Type type;
	private String data;

	public JsonObject(Type type, String data) {
		this.data = data;
		this.type = type;
	}

	public String getDataString() {
		return data;
	}

	public Type getType() {
		return type;
	}
	public JSONObject getJsonObject() throws JSONException{
		JSONObject jO = new JSONObject();
		jO.put("type", type);
		jO.put("data", data);
		return jO;
	}
}